controllers/chat/findOrCreateChannelBySignedSender.js

```diff
@@ -0,0 +1,48 @@
+const {StreamChat} = require('stream-chat');
+const {CHANNEL_TYPE_STRING, CHANNEL_TYPE} = require('../../helpers/constants');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const findOrCreateChannelBySignedSender = async (req, res) => {
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  const {members} = req.body;
+
+  try {
+    if (!members.includes(req.userId)) members.push(req.userId);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `members` array is directly modified here by pushing the `req.userId`. This can lead to unexpected behavior if the same array is used elsewhere in the code. It's better to create a new array with the additional member.
```diff
-    if (!members.includes(req.userId)) members.push(req.userId);
+    const allMembers = members.includes(req.userId) ? members : [...members, req.userId];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->